app/facades/weather_facade.rb

```diff
@@ -6,8 +6,54 @@ def self.forecast(latitude, longitude)
     weather_attr_hash = forecast_response[:properties][:periods][0].slice(:name, :temperature, :shortForecast, :detailedForecast)
     Weather.new(weather_attr_hash)
   end
-end
 
-# def weather day 1
-# def weather day 2
-# def map periods, we can idenfity by weather_object.name which day or day/night we want
+  def self.convert_forecast_data_to_weather_obj(array_of_forecast_hashes)
+    weather_objects = []
+    array_of_forecast_hashes.each do |forecast_hash|
+      forecast = forecast_hash.slice(:name, :temperature, :shortForecast, :detailedForecast)
+      weather_objects << Weather.new(forecast)
+    end
+    weather_objects
+  end
+
```

Comment:
Love the way this turned out! thanks again Stephen!

Reply:
You could do this with a map and have it set up like; `Weather.new(forecast_hash.slice(:name, :temperature, :shortForecast, :detailedForecast))`. Then you wouldn't need to do have a weather_objects array. However this looks good!